autokeras/blocks/basic.py

```diff
@@ -260,6 +261,243 @@ def _get_padding(kernel_size, output_node):
         return 'same'
 
 
+class MultiHeadSelfAttention(block_module.Block):
+    """Block for Multi-Head Self-Attention.
+
+    # Arguments
+        head_size: Int. Dimensionality of the `query`, `key` and `value` tensors
+            after the linear transformation. If left unspecified, it will be
+            tuned automatically.
+        num_heads: Int. The number of attention heads. If left unspecified,
+            it will be tuned automatically.
+    """
+
+    def __init__(self,
+                 head_size: Optional[int] = None,
+                 num_heads: Optional[int] = 8,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.head_size = head_size
+        self.num_heads = num_heads
+
+    def get_config(self):
+        config = super().get_config()
+        config.update({
+            'head_size': self.head_size,
+            'num_heads': self.num_heads})
+        return config
+
+    def build(self, hp, inputs=None):
+        """
+        # Arguments
+             hp: HyperParameters. The hyperparameters for building the model.
+             inputs: Tensor of Shape [batch_size, seq_len, embedding_dim]
+
+        # Returns
+            Self-Attention outputs of shape `[batch_size, seq_len, embedding_dim]`.
+        """
+        inputs = nest.flatten(inputs)
+        utils.validate_num_inputs(inputs, 1)
+        input_node = inputs[0]
+        shape = input_node.shape.as_list()
+        if len(shape) != 3:
+            raise ValueError(
+                'Expect the input tensor to have '
+                '3 dimensions for multi-head self-attention, '
+                'but got {shape}'.format(shape=input_node.shape))
+        # input.shape = [batch_size, seq_len, embedding_dim]
+        head_size = self.head_size or hp.Choice(
+            'head_size',
+            [32, 64, 128, 256, 512],
+            default=128)
+        num_heads = self.num_heads
+        if num_heads is None:
+            num_heads = 8
+
+        if head_size % num_heads != 0:  # how to evaluate this condition
+            raise ValueError(
+                f"embedding dimension = {head_size} should be "
+                f"divisible by number of heads = {num_heads}"
+            )
```

Comment:
Please make this  the same format as the one on 304. single quote, use .format. Also please test all the error messages in this PR in the unit tests.